gateleen-queue/src/main/java/org/swisspush/gateleen/queue/queuing/splitter/QueueSplitterImpl.java

```diff
@@ -0,0 +1,103 @@
+package org.swisspush.gateleen.queue.queuing.splitter;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpServerRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.swisspush.gateleen.core.configuration.ConfigurationResourceConsumer;
+import org.swisspush.gateleen.core.configuration.ConfigurationResourceManager;
+import org.swisspush.gateleen.queue.queuing.splitter.executors.QueueSplitExecutor;
+import org.swisspush.gateleen.queue.queuing.splitter.executors.QueueSplitExecutorFromStaticList;
+import org.swisspush.gateleen.queue.queuing.splitter.executors.QueueSplitExecutorFromRequest;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * {@inheritDoc}
+ */
+public class QueueSplitterImpl extends ConfigurationResourceConsumer implements QueueSplitter {
+
+    private final Logger log = LoggerFactory.getLogger(QueueSplitterImpl.class);
+
+    private final Map<String, Object> properties;
+
+    private boolean initialized = false;
+
+    private List<QueueSplitExecutor> queueSplitExecutors = new ArrayList<>();
+
+    public QueueSplitterImpl(
+            ConfigurationResourceManager configurationResourceManager,
+            String configResourceUri
+    ) {
+        this(configurationResourceManager, configResourceUri, new HashMap<>());
+    }
+
+    public QueueSplitterImpl(
+            ConfigurationResourceManager configurationResourceManager,
+            String configResourceUri,
+            Map<String, Object> properties
+    ) {
+        super(configurationResourceManager, configResourceUri, "gateleen_queue_splitter_configuration_schema");
+        this.properties = properties;
+    }
+
+    public Future<Void> initialize() {
+        Promise<Void> promise = Promise.promise();
+        configurationResourceManager().getRegisteredResource(configResourceUri()).onComplete((event -> {
+            if (event.succeeded() && event.result().isPresent()) {
+                initializeQueueSplitterConfiguration(event.result().get());
+                promise.complete();
+            } else {
+                log.warn("No queue splitter configuration resource with uri '{}' found. Unable to setup splitter configuration correctly", configResourceUri());
+                promise.complete();
+            }
+        }));
+        return promise.future();
+    }
+
+    public boolean isInitialized() {
```

Comment:
What do you need
```java
public boolean isInitialized() {
    return initialized;
}
```
for? Only for tests? What happens if you remove the config resource and `initialized=false` is set?

Reply:
Yes, the method is used only in QueueSplitterImpl tests to verify the current configuration is valid. Same purpose as KafkaHandler.isInitialized() also used only in KafkaHandler tests.